vault_program/src/delegate_token_account.rs

```diff
@@ -0,0 +1,42 @@
+use jito_jsm_core::loader::{
+    load_associated_token_account, load_signer, load_token_mint, load_token_program,
+};
+use solana_program::{
+    account_info::AccountInfo, entrypoint::ProgramResult, program::invoke,
+    program_error::ProgramError, pubkey::Pubkey,
+};
+use spl_token::instruction::approve;
+
+pub fn process_delegate_token_account(
+    _program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    amount: u64,
+) -> ProgramResult {
+    let [vrt_mint, token_account, owner, delegate, token_program] = accounts else {
+        return Err(ProgramError::NotEnoughAccountKeys);
+    };
+
+    load_associated_token_account(token_account, owner.key, vrt_mint.key)?;
```

Comment:
you can relax the constraints on this slightly, can be any token account (need another load_token_account method that doesn't do the ATA check)

Reply:
Added another loading

```rust
/// Loads the account as a token account, returning an error if it is not.
///
/// # Arguments
/// * `token_account` - The account to load the token account from
///
/// # Returns
/// * `Result<(), ProgramError>` - The result of the operation
pub fn load_token_account(
    token_account: &AccountInfo,
    token_program_info: &AccountInfo,
) -> Result<(), ProgramError> {
    if token_account.owner.ne(token_program_info.key) {
        msg!("Account is not owned by the token program");
        return Err(ProgramError::InvalidAccountOwner);
    }

    if token_account.data_is_empty() {
        msg!("Account data is empty");
        return Err(ProgramError::InvalidAccountData);
    }

    Ok(())
}

```